robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents Report Portal in ReportPortal
+
+    This holds the properties and function to interact with the Report Portal
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
+
+    @property
+    def api_url(self):
+        """Super url of report portal to be used as a base in all child classes
+        :returns Base url for API request
+        """
+        _api_url = f'{self.rp_url}/api/v1/{self.rp_project}/'
+        return _api_url
+
+    @property
+    def headers(self):
+        """The headers for Report Requests.
+        :returns header for API request
+        """
+        _headers = {'Authorization': f'Bearer {self.rp_api_key}'}
+        return _headers
+
+
+class Launch(ReportPortal):
+    """Represents Launch in Report Portal
+
+    This holds the properties and function to interact with the
+    Report Portal launches or specific launch
+    """
+
+    def __init__(self):
+        """Initiate report portal launch properties"""
+        super(Launch, self).__init__()
+        self.launch_url = f'{self.api_url}/launch'
+
+    @property
+    def params(self):
+        """Parameters for Launch API request
+
+        :returns dict: parameters for API launch request
+        """
+        _params = {
+            'page.page': 1,
+            'page.size': 500,
+            'page.sort': 'start_time',
+        }
+        return _params
+
+    @property
+    def launches(self):
```

Comment:
`s/launches/get_launches/`

Reply:
I dont like using these verbs along with nouns. The docstrings are good enough to understand what the def is doing.

Reply:
I think the nouns are fine without the `get_` prefix, especially in the case of properties.

That being said, see my larger comment above about the general structure here - and whether this request should be a property or a function.